### PR TITLE
Navigation: Unify `Page` component

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- removes the old `Page` component and replaces it with `PageNew`
- updates some unit tests to work correctly with the new `Page` component
- adds a unit test to make sure `initDashboard` is only ever called once by `DashboardPage` when it's wrapped in `AppChrome`
  - this prevents us loading template variables more than once
- removes last proper mentions of `topnav` from the codebase (but **not** the toggle itself, this is needed for plugins)

**Why do we need this feature?**

- T I D Y 🧹 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

File: public/app/core/components/Page/Page.tsx
Comment (joshhunt) [inline public/app/core/components/Page/Page.tsx:1]: Can we get a `core/components/Page/index.ts` export of Page. Seeing all the `import { Page } from "app/components/Page/Page"` imports is kinda wonky.

File: public/app/features/datasources/components/ButtonRow.tsx [hunk 1]
```diff
@@ -15,30 +14,14 @@ export interface Props {
   onTest: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
 }
 
-export function ButtonRow({ canSave, canDelete, onDelete, onSubmit, onTest, exploreUrl }: Props) {
+export function ButtonRow({ canSave, onSubmit, onTest, exploreUrl }: Props) {
```
Comment (joshhunt) [inline public/app/features/datasources/components/ButtonRow.tsx:17]: Do you need to update the type of Props to remove `onDelete` and `canDelete`?

**Review by joshhunt — APPROVED** [review]

**Review by torkelo — APPROVED** [review]
Nice! looks good

**Pull request merged**